tt_eager/tensor/tensor.hpp

```diff
@@ -296,6 +296,7 @@ struct Tensor {
     }
 
     const MemoryConfig memory_config() const {
+        this->wait_for_tensor_metadata_populated();
```

Comment:
This is due to poor naming on my part, but whenever we have an accessor querying the storage, we should use the `wait_for_tensor_data_populated` stall function. This function stalls until all workers for a tensor (in the multi-device case) have populated their corresponding storage attributes. I will update the naming to reflect this.